From 19374b63b213b2807290d2c915e88fbd7f4a8e14 Mon Sep 17 00:00:00 2001
From: GuaiYiHu <guaiyihu@foxmail.com>
Date: Thu, 15 Oct 2026 00:31:38 +0800
Subject: [PATCH] Bluetooth: skip LE identity generation on classic controllers

Preserve link-key registration without submitting unsupported LE Rand commands on BCM4329.

Adapted-from: 3b695d3d1d7c1b2c1193b33fa51537bc8619c190
Change-Id: I535559d9c220b558973dd8ef31232cf527ba1dd7
---
 system/stack/btm/btm_sec.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/system/stack/btm/btm_sec.cc b/system/stack/btm/btm_sec.cc
index 4b18529520..47563b2b33 100644
--- a/system/stack/btm/btm_sec.cc
+++ b/system/stack/btm/btm_sec.cc
@@ -396,7 +396,8 @@ bool BTM_SecRegister(const tBTM_APPL_INFO* p_cb_info) {
     SMP_Register(btm_proc_smp_cback);
     Octet16 zero{0};
     /* if no IR is loaded, need to regenerate all the keys */
-    if (btm_sec_cb.devcb.id_keys.ir == zero) {
+    if (controller_get_interface()->supports_ble() &&
+        btm_sec_cb.devcb.id_keys.ir == zero) {
       btm_ble_reset_id();
     }
   } else {
